### PR TITLE
[dev] Add Filter Drawer Component

### PR DESCRIPTION
**Pull request by ajaybushel**
Here's how it looks:

Simulator Screen Recording - iPhone 13 - 2022-06-21 at 12 10 59

### CODE REVIEW

**Review by raoufr — APPROVED** [review]
Looking good to me
However, I would like to discuss the dependency on the Icons library and whether this might have long term implications. I will bring it up on our weekly meeting to discuss. thanks

File: src/modules/filter-drawer/filter-drawer.tsx [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+          </AnimatedWrapper>
+        </SectionWrapper>
+
+        {/* Close Button */}
+        <AnimatedCloseWrapper style={{opacity: closeButtonOpacity}}>
+          <TouchableOpacity
+            testID="drawer-close-button"
+            disabled={shouldShowSecondaryContent}
+            onPress={onCloseModal}>
+            <Icon file={IconCloseXCircle} />
```
Comment (krishnakant-technossus) [inline src/modules/filter-drawer/filter-drawer.tsx:340]: please increase the size of close icon upto 32px. and also make top and right margins to 11px.

File: src/modules/filter-drawer/filter-drawer.tsx [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+            <PrimaryContentWrapper testID="drawer-primary-content">
+              <ScrollView showsVerticalScrollIndicator={false}>
+                {/* Header */}
+                <Header.Wrapper>
+                  <Animated.View
+                    style={{
+                      opacity: headerContentOpacity,
+                      transform: [{translateX: headerContentTranslateX}],
+                    }}>
+                    <Header.Title variant="h2">Filters</Header.Title>
```
Comment (krishnakant-technossus) [inline src/modules/filter-drawer/filter-drawer.tsx:262]: Please make sure that filters title should have top margin of 33px. and can you also make this title prop dynamic?

File: src/modules/filter-drawer/filter-drawer.tsx [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+                        onPress={onCancel}>
+                        Cancel
+                      </Button>
+                    )}
+
+                    <Button
+                      color="success"
+                      testID="drawer-apply-filters-button"
+                      onPress={onApplyFilters}>
+                      Apply Filters
```
Comment (krishnakant-technossus) [inline src/modules/filter-drawer/filter-drawer.tsx:297]: 'Apply filters' button title should be dynamic and should have visibility boolean flag. So consumer can show/hide if filters selected or not.

File: src/modules/filter-drawer/filter-drawer.tsx [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+        {/* Sections */}
+        <SectionWrapper>
+          {/* Content */}
+          <AnimatedWrapper
+            style={{transform: [{translateX: contentWrapperTranslateX}]}}>
+            {/* Primary Content */}
+            <PrimaryContentWrapper testID="drawer-primary-content">
+              <ScrollView showsVerticalScrollIndicator={false}>
+                {/* Header */}
+                <Header.Wrapper>
```
Comment (krishnakant-technossus) [inline src/modules/filter-drawer/filter-drawer.tsx:256]: Make the header wrapper sticky. It should not scroll with the content.

Comment (ajaybushel) [reply src/modules/filter-drawer/filter-drawer.tsx:256]: If we are going to follow the ME-like design for Drawers, the headers don't stick. Only the close button is fixed on the top right. Can you please confirm this with James?

Comment (krishnakant-technossus) [reply src/modules/filter-drawer/filter-drawer.tsx:256]: Okay, that is fine. ignore this.

**Comment by ajaybushel** [comment]
Looks good to me.

**Pull request merged**